src/SourceLink.GitLab/GetSourceLinkUrl.cs

```diff
@@ -16,7 +16,34 @@ public sealed class GetSourceLinkUrl : GetSourceLinkUrlGitTask
         protected override string HostsItemGroupName => "SourceLinkGitLabHost";
         protected override string ProviderDisplayName => "GitLab";
 
+        private const string VersionMetadataName = "Version"; // TODO rename to GitLabVersion? Or leave it as Version to be similar with Microsoft.SourceLink.Bitbucket.Git.GetSourceLinkUrl?
+        private static readonly Version s_versionWithNewUrlFormat = new Version(13, 5);
+
         protected override string? BuildSourceLinkUrl(Uri contentUri, Uri gitUri, string relativeUrl, string revisionId, ITaskItem? hostItem)
-            => UriUtilities.Combine(UriUtilities.Combine(contentUri.ToString(), relativeUrl), "raw/" + revisionId + "/*");
+        {
+            var path = GetVersion(hostItem) >= s_versionWithNewUrlFormat
+                ? "-/raw/" + revisionId + "/*"
+                : "raw/" + revisionId + "/*";
+            return UriUtilities.Combine(UriUtilities.Combine(contentUri.ToString(), relativeUrl), path);
+        }
+
+        private Version GetVersion(ITaskItem? hostItem)
+        {
+            // TODO get GitLab version from the environment variable CI_SERVER_VERSION?
```

Comment:
>TODO [](http://example.com/codeflow?start=15&length=4)

I don't think we should use CI_ variables directly since they are only available on CI build and not locally.

Reply:
I think `CI_SERVER_VERSION` is GitLab specific. My reasoning was that since we are using GitLab, we are also probably using GitLab CI. And in that case, when the user did not specify a Version, it would be deduced here. But maybe that too many assumptions?

Reply:
I think it's better to not introduce implicit dependency on CI specific environment variables. It's easy to set the `Version` metadata to `CI_SERVER_VERSION` in .props file if the user of the package wants to do that. We can document this option.